network/endpoint.go

```diff
@@ -386,3 +387,25 @@ func (epInfo *EndpointInfo) IsEndpointStateIncomplete() bool {
 	}
 	return false
 }
+
+func (ep *endpoint) validateEndpoint() error {
+	if ep.ContainerID == "" || ep.NICType == "" {
+		return errors.New("endpoint struct must contain a container id and nic type")
+	}
+	return nil
+}
+
+func validateEndpoints(eps []*endpoint) error {
```

Comment:
should we add some checks on updateEndpoint() side as well?

Reply:
for IB, we are passing following fields:
        Id:          epInfo.MasterIfName,  (not a must)
        IfName:      epInfo.MasterIfName, (not a must)
        ContainerID: epInfo.ContainerID,
        MacAddress:  epInfo.MacAddress,
        NICType:     cns.BackendNIC,